test/test_fourier.jl

```diff
@@ -0,0 +1,419 @@
+function compare_tables(table1, table2; rtol=0.001, discard = [])
+
+    s_discard = Symbol.(discard)
+    test_result = true
+
+    for key in propertynames(table1)
+        if key in s_discard
+            continue
+        end
+        oe, oc = getproperty(table1,key), getproperty(table1,key)
+        if oe isa Integer || oe isa String
+            if !(oe==oc) 
+                test_result = false
+                break
+            end
+        elseif isnothing(oe)
+            if !(isnothing(oc)) 
+                test_result = false
+                break
+            end
+        else
+            if !(≈(oe,oc,rtol=rtol)) 
+                test_result = false
+                break
+            end
+        end
+    end
+
+    table1 = Metadata.drop_metadata(table1)
+    table2 = Metadata.drop_metadata(table2)
+
+    for field in names(table1)
+        if field in discard
+            continue
+        end
+        oe, oc = table1[!,field], table2[!,field]
+
+        if !(≈(oe,oc,rtol=rtol)) 
+            test_result = false
+            break
+        end
+    end
+    @test test_result
+end
+
+@testset "positive_fft_bins" begin
+    freq = fftfreq(11)
+    good = freq .> 0
+    goodbins = positive_fft_bins(11)
+    @test freq[good]==freq[goodbins]
+end
+
+@testset "test_coherence" begin
+
+    fid = h5open(joinpath(@__DIR__ ,"data","sample_variable_lc.h5"),"r")
+    data = (fid["dataset1"][1:10000]) .* 1000.0
+    close(fid)
+
+    data1 = rand.(Poisson.(data))
+    data2 = rand.(Poisson.(data))
+    ft1 = fft(data1)
+    ft2 = fft(data2)
+    dt = 0.01
+    N = length(data)
+    mean = Statistics.mean(data)
+    meanrate = mean / dt
+    freq = fftfreq(length(data), 1/dt)
+    good = (freq .> 0 .&& freq .< 0.1)
+    ft1, ft2 = ft1[good], ft2[good]
```

Comment:
This is more idiomatic:
```suggestion
    good = 0 .< freq .< 0.1
    keepat!(ft1, good)
    keepat!(ft2, good)
```